src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -599,4 +660,604 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(
+                                          &object_type_ref);
+  if (FAILED(hr)) {
+    Warn("GenerateVoidILStartupMethod: DefineTypeRefByName failed");
+    return hr;
+  }
+
+  // Define a new TypeDef __DDVoidMethodType__ that extends System.Object
+  mdTypeDef new_type_def;
+  hr = metadata_emit->DefineTypeDef("__DDVoidMethodType__"_W.c_str(), tdAbstract | tdSealed,
+                               object_type_ref, NULL, &new_type_def);
+  if (FAILED(hr)) {
+    Warn("GenerateVoidILStartupMethod: DefineTypeDef failed");
+    return hr;
+  }
+
+  // Define a new static method __DDVoidMethodCall__ on the new type that has a void return type and takes no arguments
+  BYTE initialize_signature[] = {
+    IMAGE_CEE_CS_CALLCONV_DEFAULT, // Calling convention
+    0,                             // Number of parameters
+    ELEMENT_TYPE_VOID,             // Return type
+    ELEMENT_TYPE_OBJECT            // List of parameter types
+  };
+  hr = metadata_emit->DefineMethod(new_type_def,
+                              "__DDVoidMethodCall__"_W.c_str(),
+                              mdStatic,
+                              initialize_signature,
+                              sizeof(initialize_signature),
+                              0,
+                              0,
+                              ret_method_token);
+  if (FAILED(hr)) {
+    Warn("GenerateVoidILStartupMethod: DefineMethod failed");
+    return hr;
+  }
+
+  // Define a method on the managed side that will PInvoke into the profiler method:
+  // C++: void GetAssemblyAndSymbolsBytes(BYTE** pAssemblyArray, int* assemblySize, BYTE** pSymbolsArray, int* symbolsSize)
+  // C#: static extern void GetAssemblyAndSymbolsBytes(out IntPtr assemblyPtr, out int assemblySize, out IntPtr symbolsPtr, out int symbolsSize)
+  mdMethodDef pinvoke_method_def;
+  COR_SIGNATURE get_assembly_bytes_signature[] = {
+      IMAGE_CEE_CS_CALLCONV_DEFAULT, // Calling convention
+      4,                             // Number of parameters
+      ELEMENT_TYPE_VOID,             // Return type
+      ELEMENT_TYPE_BYREF,            // List of parameter types
+      ELEMENT_TYPE_I,
+      ELEMENT_TYPE_BYREF,
+      ELEMENT_TYPE_I4,
+      ELEMENT_TYPE_BYREF,
+      ELEMENT_TYPE_I,
+      ELEMENT_TYPE_BYREF,
+      ELEMENT_TYPE_I4,
+  };
+  hr = metadata_emit->DefineMethod(
+      new_type_def, "GetAssemblyAndSymbolsBytes"_W.c_str(), mdStatic | mdPinvokeImpl | mdHideBySig,
+      get_assembly_bytes_signature, sizeof(get_assembly_bytes_signature), 0, 0,
+      &pinvoke_method_def);
+  if (FAILED(hr)) {
+    Warn("GenerateVoidILStartupMethod: DefineMethod failed");
+    return hr;
+  }
+
+  metadata_emit->SetMethodImplFlags(pinvoke_method_def, miPreserveSig);
+  if (FAILED(hr)) {
+    Warn("GenerateVoidILStartupMethod: SetMethodImplFlags failed");
+    return hr;
+  }
+
+  mdModuleRef profiler_ref;
+  hr = metadata_emit->DefineModuleRef("DATADOG.TRACE.CLRPROFILER.NATIVE.DLL"_W.c_str(),
```

Comment:
`L"DATADOG.TRACE.CLRPROFILER.NATIVE.DLL"`
Also, for the *nix pieces, the extension changes, so this will need to be from an #ifdef probably.

Reply:
Yep you're right. This method is not called on *nix currently so I didn't include it here, but I think we would just need to remove the ".dll" extension